src/app/public/src/modules/link/link.directive.ts

```diff
@@ -0,0 +1,24 @@
+import { Directive, HostListener, Input } from '@angular/core';
+
+import { StacheNavService } from '../nav';
+
+@Directive({
+  selector: '[stacheRouterLink]'
+})
+export class StacheRouterLinkDirective {
+  @Input()
+  public stacheRouterLink: string;
+
+  @Input()
+  public fragment: string;
+
+  constructor(private navService: StacheNavService) { }
+
+  @HostListener('click')
+  public navigate(): void {
```

Comment:
Needs an `event.preventDefault();` so that normal anchor linking behavior will be ignored. To test, add `href="#"` to your demo page.